### PR TITLE
Parse json comment in new API.

### PR DESCRIPTION
**Pull request by ailzhang**
Companion PR: https://github.com/pytorch/pytorch/pull/33908

### CODE REVIEW

File: scripts/gen.py [hunk 1]
```diff
@@ -947,18 +959,27 @@ def generate_registrations(fgens, overrides):
   return code + ';\n}\n', overridden
 
 
+# XLA is only able to override non-compound ops and whitelisted compound ops.
+def is_overrideable(fgen):
+  if not fgen.compound:
+      return True
+  if fgen.mapsig in _FN_WHITELIST or fgen.func in _FN_WHITELIST:
```
Comment (dlibenzi) [inline scripts/gen.py]: return ...

File: scripts/gen.py [hunk 2]
```diff
@@ -880,7 +889,8 @@ def extract_functions(path):
     fndef = m.group(1)
     try:
       _XPARSER.parse(fndef)
-      functions.append(FuncDef(cpp_sig=fndef, aten_sig=m.group(2)))
+      comment = json.loads(m.group(2))
+      functions.append(FuncDef(cpp_sig=fndef, aten_sig=comment['schema'], compound=comment['compound']=='1'))
```
Comment (dlibenzi) [inline scripts/gen.py]: I would make an API:
```Python
def _create_funcdef(fndef, jdata):
  ...
  return FuncDef(cpp_sig=fndef, ...)
```
That creates a `FuncDef` out of the `fndef` and JSON data.

File: scripts/gen.py [hunk 3]
```diff
@@ -24,11 +25,11 @@ class ArgTemplate(string.Template):
   idpattern = r'[a-z0-9_]+'
 
 
-FuncDef = namedtuple_with_defaults('FuncDef', 'cpp_sig, aten_sig')
+FuncDef = namedtuple_with_defaults('FuncDef', 'cpp_sig, aten_sig, compound')
```
Comment (dlibenzi) [inline scripts/gen.py]: I am debating whether we should expand the JSON data or just stick it here as a whole (that excludes aten_sig).

File: scripts/gen.py [hunk 4]
```diff
@@ -24,11 +25,11 @@ class ArgTemplate(string.Template):
...
 
 FuncGen = namedtuple_with_defaults(
     'FuncGen',
-    'tree, xtree, rwxtree, func, xfunc, code, sig, rwsig, cppsig, funsig, mapsig, aten_sig'
+    'tree, xtree, rwxtree, func, xfunc, code, sig, rwsig, cppsig, funsig, mapsig, aten_sig, compound'
```
Comment (dlibenzi) [inline scripts/gen.py]: I want a better name than compound 😉

File: scripts/gen.py (hunk 1 above)
Comment (ailzhang) [reply scripts/gen.py]: To double check, is this about the 4 spaces before return? ;)

File: scripts/gen.py (hunk 3 above)
Comment (ailzhang) [reply scripts/gen.py]: I feel it's good to have them in separate fields given that they're indeed func properties?

File: scripts/gen.py (hunk 1 above)
Comment (dlibenzi) [reply scripts/gen.py]: Just return the value of that `if` conditional.

File: scripts/gen.py (hunk 3 above)
Comment (dlibenzi) [reply scripts/gen.py]: SG

File: scripts/gen.py [hunk 5]
```diff
@@ -873,14 +882,19 @@ def is_tensor_api(fndef):
 def extract_functions(path):
   functions = []
   errors = []
+
+  def _create_funcdef(fndef, jdata):
```
Comment (dlibenzi) [inline scripts/gen.py]: No need for this to be nested.

File: scripts/gen.py [hunk 6]
```diff
@@ -870,17 +879,23 @@ def is_tensor_api(fndef):
   return m is not None, fndef
 
 
+def create_funcdef(fndef, jdata):
+  fields = json.loads(jdata)
+  return FuncDef(cpp_sig=fndef, aten_sig=fields['schema'], leaf=fields['compound'] == '0')
```
Comment (dlibenzi) [inline scripts/gen.py]: I feel it is more robust not to assume presence of given metadata.
So maybe this?
```Python
fileds.get('compound', '0') == '0'
```

**Pull request merged**